web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx

```diff
@@ -186,6 +192,20 @@ export function ClusterResources({
       ]
     ),
   });
+  const { samlAppToDelete } = useSamlAppAction();
+  if (samlAppToDelete?.backendDeleted) {
+    const index = resources.findIndex(e => {
+      switch (e.kind) {
+        case 'app':
+          return e.name === samlAppToDelete.name && e.samlApp;
+        default:
+          return false;
+      }
+    });
+    if (index >= 0) {
+      resources.splice(index, 1);
```

Comment:
`Array.prototype.splice` mutates the array. In React, you almost never want to mutate state. https://react.dev/learn/updating-objects-in-state#local-mutation-is-fine

Instead, you can filter it instead:

<details>
<summary>Patch</summary>

```diff
diff --git a/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx b/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx
index 472bbca42c..cc49985411 100644
--- a/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx
+++ b/web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx
@@ -16,7 +16,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, { useCallback, useState } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 
 import { Flex } from 'design';
 import { Danger } from 'design/Alert';
@@ -155,7 +155,12 @@ export function ClusterResources({
     getClusterPinnedResources: getCurrentClusterPinnedResources,
   };
 
-  const { fetch, resources, attempt, clear } = useUnifiedResourcesFetch({
+  const {
+    fetch,
+    resources: unfilteredResources,
+    attempt,
+    clear,
+  } = useUnifiedResourcesFetch({
     fetchFunc: useCallback(
       async (paginationParams, signal) => {
         const response = await teleCtx.resourceService.fetchUnifiedResources(
@@ -192,20 +197,22 @@ export function ClusterResources({
       ]
     ),
   });
+
   const { samlAppToDelete } = useSamlAppAction();
-  if (samlAppToDelete?.backendDeleted) {
-    const index = resources.findIndex(e => {
-      switch (e.kind) {
-        case 'app':
-          return e.name === samlAppToDelete.name && e.samlApp;
-        default:
-          return false;
-      }
-    });
-    if (index >= 0) {
-      resources.splice(index, 1);
-    }
-  }
+  const resources = useMemo(
+    () =>
+      samlAppToDelete?.backendDeleted
+        ? unfilteredResources.filter(
+            res =>
+              !(
+                res.kind === 'app' &&
+                res.samlApp &&
+                res.name === samlAppToDelete.name
+              )
+          )
+        : unfilteredResources,
+    [samlAppToDelete, unfilteredResources]
+  );
 
   // This state is used to recognize when the `params` value has changed,
   // and reset the overall state of `useUnifiedResourcesFetch` hook. It's tempting to use a

```
</details>

Reply:
True, I definitely overlooked this one from poc to PR updated with your patch in https://github.com/gravitational/teleport/pull/44311/commits/774df9aa5f3c55150628661d0d5d25b7d9a48dec